service/controller/v5/resource/blobobject/blob_client.go

```diff
@@ -0,0 +1,119 @@
+package blobobject
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2018-07-01/storage"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+)
+
+const (
+	blobFormatString      = `https://%s.blob.core.windows.net`
+	resourceNotFoundError = "NotFound"
+	maxRetriesRequests    = 3
+)
+
+type BlobClient struct {
+	storageAccountsClient *storage.AccountsClient
+	containerURL          azblob.ContainerURL
+}
+
+func (c *BlobClient) BlobExists(ctx context.Context, blobName string) (bool, error) {
```

Comment:
This is exported and not even used?

Reply:
It is exported as I expect this to be used in `instance` resource so that we can stop reconciling instance resource if blob with ignition is not ready yet